### ISSUE TITLE
Add title attribute to YouTube iframes

### ISSUE THREAD

**Issue by jrrdnx** [spicywebau/craft-embedded-assets#164]
For accessibility, iframes should have a title attribute. I see that the Vimeo iframe includes this, but presume this is returned by Vimeo. Is it possible to add this attribute to the YouTube iframe if it's not included from YouTube already? Possibly using the Asset's title as a fallback?

- Embedded Assets version: 2.5.1
- Craft CMS version: 3.6.8

**Comment by ttempleton** [comment]
Not really a fan of forcibly adding iframe attributes the provider doesn't include -- ideally they'd include the title attribute and the accessibility issue would be fixed for everyone, not just Embedded Assets users.  Instead, we'll make plans to allow users to specify their own iframe attribute overrides, which could be used to add any other iframe attributes that are needed for any reason.

**Comment by ccchapman** [comment]
You can use Craft's `|attr` [filter](https://craftcms.com/docs/3.x/dev/filters.html#attr) to accomplish the override suggested above.

```twig
{{ craft.embeddedAssets.get(asset).getVideoCode()|attr({
  title: asset.title,
}) }}
```

**Comment by jrrdnx** [comment]
The only downside is that it wouldn't be using the title of the video as it is set on YouTube.

+1 for the `|attr` filter suggestion, as we can always use the Asset's title since that's available. Would just prefer to use YouTube's video title.

**Comment by ttempleton** [comment]
In that case, to tweak @ccchapman's example a bit, you should be able to do something like this:

```twig
{% set embeddedAsset = craft.embeddedAssets.get(asset) %}
{{ embeddedAsset.code|attr({
  title: embeddedAsset.title,
}) }}
```

The embedded asset's `title` is the title that's given to us by YouTube, so that should be ideal for your use case.

Thanks for the great `|attr` suggestion @ccchapman, that probably makes it unnecessary to implement that in the plugin directly so will close this issue.
